Tools/Scripts/webkitdirs.pm

```diff
@@ -617,16 +628,46 @@ sub determineXcodeDestination
     }
         
     if (!$generic && $xcodeSDKPlatformName =~ /simulator$/) {
+        # Two goals:
+        # 1. Find a simulator device to build for, to avoid building multiple architectures.
+        # 2. Try to pick a simulator that's been used before (either by command-line or IDE builds) to avoid
+        #    unnecessary recompilation.
+        #
+        # Changing the targeted simulator between builds breaks incremental building, because it changes
+        # build settings like TARGET_DEVICE_IDENTIFIER.
+        my $prevBuildRequest = first {
+            $_->{parameters}->{activeRunDestination}->{platform} eq $xcodeSDKPlatformName
+        } xcodeBuildRequestsInRecencyOrder();
+        my $prevUDID = $prevBuildRequest->{parameters}->{overrides}->{synthesized}->{table}->{TARGET_DEVICE_IDENTIFIER} if $prevBuildRequest;
+        if ($prevBuildRequest && !$prevUDID) {
+            warn "Can't find UDID in previous $xcodeSDKPlatformName build request, builds may not be incremental.\n";
+        }
+        
+        my @devices = iOSSimulatorDevices();
+        my $prevDevice = first { $prevUDID && $_->{UDID} eq $prevUDID } @devices;
+        if ($prevUDID && !$prevDevice) {
+            warn "Simulator with UDID '$prevUDID' not found, falling back to another available simulator. " .
```

Comment:
I wonder if these should be more noticeable (with extra blank lines perhaps). Or are they already printed early enough that they are noticeable?

Reply:
They would usually be the first lines printed after running `make`, because they show up before xcodebuild is invoked. But I'm open to adding more formatting / indication that these are WebKit-specific messages and not xcodebuild noise.

Reply:
Going to hold off on this, I'd rather explore it in a follow-up where we refactor `set-webkit-configuration` to share some more UI code with webkitdirs itself.

> They would usually be the first lines printed after running `make`, because they show up before xcodebuild is invoked.

Correction: it's the first line after the "===== BUILDING <scheme> =====" message. I still think this is high up enough to be noticeable for someone investigating why their build is slow.